eclipse-collections/src/main/java/org/eclipse/collections/impl/map/strategy/mutable/UnifiedMapWithHashingStrategy.java

```diff
@@ -1079,6 +1080,22 @@ public V removeKey(K key)
         return this.remove(key);
     }
 
+    @Override
+    public boolean removeAllKeys(Set<? extends K> keys)
+    {
+        Objects.requireNonNull(keys);
+        int previousSize = this.size();
+        if (this.size() > keys.size())
+        {
+            keys.forEach(this::removeKey);
+        }
+        else
+        {
+            this.keySet().removeIf(keys::contains);
+        }
+        return previousSize != this.size();
+    }
```

Comment:
Unrelated to the current change, but we're probably getting to the point where there is enough commonality between UnifiedMap and UnifiedMapWithHashingStrategy that we could extract an AbstractUnifiedMap. I think we talked about doing it in the past but we were worried about performance. If no state were pulled up and it were just used for "default methods" like these, I think it would be fine.

Reply:
Raised #698